10_code/threshold.py

```diff
@@ -143,72 +148,92 @@
 pop_counties_new[["County", "State"]] = pop_counties_new.NAME.str.split(",", n=1, expand=True)
 pop_counties_new["State"] = pop_counties_new.State.str.strip()
 
-pop_new = pop_counties_new[(pop_counties_new["State"].isin(group_tx_names))]
+pop_new = pop_counties_new[(pop_counties_new["State"].isin(texas_names))]
 
 pop_tx_new = pop_new[["State", "County", "year", "population"]]
 
-pop_tx_new["year"].value_counts() #884 counties ✅
+pop_tx_new["year"].value_counts() 
 
-assert sorted(pop_tx_new["State"].unique()) == sorted(group_tx_names)
+assert sorted(pop_tx_new["State"].unique()) == sorted(texas_names)
 
 #merge to make a texas population dataset
 pop_tx_merged = pd.concat([pop_tx_old, pop_tx_new])
 pop_tx_merged.sort_values(by = ["State", "County"], inplace=True)
 
+#introduce fips codes to population data
+fips_tx = fips[(fips["state"].isin(texas_states))].copy()
+fips_tx.rename(columns={"name" : "County"}, inplace=True)
+#dictionary to map state names as abbreviation in population data
+states_TX = {
+    "Texas": "TX",
+    "Utah": "UT",
+    "Georgia": "GA",
+    "Colorado": "CO",
+    "California": "CA",
+    "North Dakota": "ND",
+    "Illinois": "IL",
+    "Louisiana": "LA",
+    "Maryland": "MD",
+    "Oklahoma": "OK",
+}
+#mapping abbreviations
+pop_tx_merged['State'] =pop_tx_merged['State'].map(states_TX)
+pop_tx_merged.rename(columns={"State":"state"}, inplace=True)
 
 #introduce fips codes to population data
-fips_tx = fips[(fips["state"].isin(group_tx))].copy()
+fips_tx = fips[(fips["state"].isin(texas_states))].copy()
 fips_tx.rename(columns={"name" : "County"}, inplace=True)
 
 #add fips to population
-pop_tx_fips = pd.merge(fips_tx, pop_tx_merged, on = "County", how = "left", indicator=True)
+pop_tx_fips = pd.merge(fips_tx, pop_tx_merged, on = ["state","County"], how = "left", indicator=True)
 pop_tx_fips["_merge"].value_counts() #✅ both 
 pop_tx_final = pop_tx_fips[["state", "fips", "County", "year", "population"]].copy()
 
 #reintroduce overdose data
 
-overdose_tx = deaths_tx[(~(deaths_tx["Year"] == 2015))].copy()
-overdose_tx.rename(columns={"County Code": "fips","State":"state"}, inplace=True)
-overdose_tx["fips"] = overdose_tx["fips"].astype(np.int64)
-overdose_tx["year"] = overdose_tx["Year"].astype(np.int64)
-overdose_tx.sort_values(by = ["state", "fips"], inplace=True)
+overdose_deaths_tx = deaths_tx[(~(deaths_tx["Year"] == 2015))].copy()
+overdose_deaths_tx.rename(columns={"County Code": "fips","State":"state"}, inplace=True)
```

Comment:
I think we should do this at the top of the file and move forward with a standardized naming convention for these columns. 🥲 guilty of this too